plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/cache/TestCachingHiveMetastoreWithQueryRunner.java

```diff
@@ -142,18 +148,37 @@ public void testIllegalFlushHiveMetastoreCacheProcedureCalls()
         var illegalParameterMessage = "Illegal parameter set passed. ";
         var validUsageExample = "Valid usages:\n - 'flush_metadata_cache()'\n - flush_metadata_cache(schema_name => ..., table_name => ..., partition_column => ARRAY['...'], partition_value => ARRAY['...'])";
 
-        assertThatThrownBy(() -> queryRunner.execute("CALL system.flush_metadata_cache('dummy_schema')"))
+        assertThatThrownBy(() -> getQueryRunner().execute("CALL system.flush_metadata_cache('dummy_schema')"))
                 .hasMessage("Procedure should only be invoked with named parameters. " + validUsageExample);
 
-        assertThatThrownBy(() -> queryRunner.execute("CALL system.flush_metadata_cache(schema_name => 'dummy_schema')"))
+        assertThatThrownBy(() -> getQueryRunner().execute("CALL system.flush_metadata_cache(schema_name => 'dummy_schema')"))
                 .hasMessage(illegalParameterMessage + validUsageExample);
-        assertThatThrownBy(() -> queryRunner.execute("CALL system.flush_metadata_cache(schema_name => 'dummy_schema', table_name => 'dummy_table')"))
+        assertThatThrownBy(() -> getQueryRunner().execute("CALL system.flush_metadata_cache(schema_name => 'dummy_schema', table_name => 'dummy_table')"))
                 .hasMessage(illegalParameterMessage + validUsageExample);
 
-        assertThatThrownBy(() -> queryRunner.execute("CALL system.flush_metadata_cache(schema_name => 'dummy_schema', table_name => 'dummy_table', partition_column => ARRAY['dummy_partition'])"))
+        assertThatThrownBy(() -> getQueryRunner().execute("CALL system.flush_metadata_cache(schema_name => 'dummy_schema', table_name => 'dummy_table', partition_column => ARRAY['dummy_partition'])"))
                 .hasMessage("Parameters partition_column and partition_value should have same length");
     }
 
+    @Test
+    public void testPartitionAppend()
+    {
+        getQueryRunner().execute("CREATE TABLE test_part_append " +
+                "(name varchar, partkey varchar) " +
+                "WITH (partitioned_by = ARRAY['partkey'])");
+
+        String row = "('some name', 'part1')";
+
+        // if metastore caching was enabled on workers than any worker which tries to INSERT into same partition twice
+        // will fail because it would've cached the absence of the partition
+        for (int i = 0; i < nodeCount + 1; i++) {
```

Comment:
Maybe we should `verify` the number of nodes query runner is running with to ensure this test's assumptions remain valid.

Reply:
Adde verification in first test line.